pkg/cloud/cloud.go

```diff
@@ -0,0 +1,37 @@
+package cloud
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+type Cloud interface {
+	GetMetadata() MetadataService
+}
+
+type cloud struct {
+	metadata MetadataService
+}
+
+// NewCloud returns a new instance of AWS cloud
+// It panics if session is invalid
+func NewCloud() (Cloud, error) {
+	sess := session.Must(session.NewSession(&aws.Config{}))
```

Comment:
kinda weird to return error but also have a panic, i would either change the name to `MustNewCloud` or `recover` and propagate the error if `session.Must` panics